src/gt4py/next/iterator/transforms/collapse_tuple.py

```diff
@@ -80,7 +86,9 @@ def visit_FunCall(self, node: ir.FunCall, **kwargs) -> ir.Node:
                     # tuple argument differs, just continue with the rest of the tree
                     return self.generic_visit(node)
 
-            if self.ignore_tuple_size or _get_tuple_size(first_expr) == len(node.args):
+            if self.ignore_tuple_size or _get_tuple_size(self._node_types[id(first_expr)]) == len(
```

Comment:
I don't think that I want `self._node_types[id(first_expr)]` to become a pattern. Any thoughts?

Reply:
When you say annex in this context, you mean we guarantee that the type is always updated/preserved in all passes, right? Otherwise we would also have to run it before each pass.
Independent of the performance problem, is there any disadvantage in using a pass-local annex here?

Reply:
With respect to the annex being "dangerous" yes. If it's only pass local you can still screw things up, but it is easier to find the problem (i.e. you do a transformation of the IR and some of the newly created nodes don't have a type or the type changed) as it is local to the pass.

Reply:
See also this commit on how the code looks like if we use the annex: https://github.com/GridTools/gt4py/pull/1260/commits/a09f18f1d227c9a557a6930b7aa17fd897d08294